### PR TITLE
Aof rewrite wipe

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -54,5 +54,21 @@ int ReadConfig(RedisModuleString **argv, int argc, const char **err) {
     RMUtil_ParseArgsAfter("FRISOINI", argv, argc, "c", &RSGlobalConfig.frisoIni);
   }
 
+  int onTmoIndex = RMUtil_ArgIndex("ON_TIMEOUT", argv, argc);
```
Comment (dvirsky) [inline src/config.c]: you do know that there's a util for that, right? RMUtil_ParseArgAfter or something.

Comment (dvirsky) [reply src/config.c]: it would just simplify this code and have the same effect.

File: src/pytest/test.py [hunk 2]
```diff
@@ -1405,98 +1405,6 @@ def testInfoCommand(self):
                 for option in filter(None, combo):
                     self.assertTrue(option in opts)
 
-    def aofTestCommon(self, reloadfn):
```
Comment (dvirsky) [inline src/pytest/test.py]: This means we have no test for aof with rdb preamble. How do we test that scenario?

File: src/query_plan.c [hunk 3]
```diff
@@ -138,7 +161,8 @@ void Query_OnReopen(RedisModuleKey *k, void *privdata) {
                            (now.tv_nsec - q->execCtx.startTime.tv_nsec);
     // printf("Elapsed: %zdms\n", durationNS / 1000000);
     // Abort on timeout
-    if (durationNS > RSGlobalConfig.queryTimeoutMS * 1000000) {
+    if (durationNS > q->opts.timeoutMS * 1000000) {
+      printf("Setting timed out state!\n");
```
Comment (dvirsky) [inline src/query_plan.c]: remove printf - convert to log if we have a context here

File: src/query_plan.c [hunk 4]
```diff
@@ -179,6 +203,14 @@ QueryPlan *Query_BuildPlan(RedisSearchCtx *ctx, QueryParseCtx *parsedQuery, RSSe
...
   plan->conc = opts->concurrentMode ? malloc(sizeof(*plan->conc)) : NULL;
   plan->opts = opts ? *opts : RS_DEFAULT_SEARCHOPTS;
+  if (plan->opts.timeoutMS == 0) {
+    plan->opts.timeoutMS = RSGlobalConfig.queryTimeoutMS;
+  }
+  if (plan->opts.timeoutPolicy == TimeoutPolicy_Default) {
+    plan->opts.timeoutPolicy = RSGlobalConfig.timeoutPolicy;
+  }
+
+  // printf("Timeout policy: %d\n", plan->opts.timeoutPolicy);
```
Comment (dvirsky) [inline src/query_plan.c]: remove

File: src/result_processor.h [hunk 5]
```diff
@@ -132,6 +132,9 @@ static inline RSValue *SearchResult_GetValue(SearchResult *res, RSSortingTable *
 // EOF - no results from this processor
 #define RS_RESULT_EOF 2
 
+// STOP - Abort processing immediately!
+#define RS_RESULT_STOP -1
```
Comment (dvirsky) [inline src/result_processor.h]: Remove this please. we are assuming that we either have ok/eof/err on so many places, do not introduce another return code. use the query state instead.

File: src/query_plan.c [hunk 6]
```diff
@@ -80,19 +80,35 @@ static size_t serializeResult(QueryPlan *qex, SearchResult *r, RSSearchFlags fla
   return count;
 }
 
+/**
+ * Returns true if the query has timed out and the user has requested
+ * that we do not drain partial results.
+ */
+static int hasNoDrainTimeout(const QueryPlan *qex) {
```
Comment (dvirsky) [inline src/query_plan.c]: convert to a macro, this is in a very tight loop.

File: src/query_plan.c [hunk 7]
```diff
@@ -80,19 +80,35 @@ static size_t serializeResult(QueryPlan *qex, SearchResult *r, RSSearchFlags fla
   return count;
 }
 
+/**
+ * Returns true if the query has timed out and the user has requested
+ * that we do not drain partial results.
+ */
```
Comment (dvirsky) [inline src/query_plan.c]: the name is not clear. should be TIMEOUT_FAIL or something like that.

File: src/result_processor.h (hunk 5 above)
Comment (mnunberg) [reply src/result_processor.h]: yeah, this is a leftover.

File: src/pytest/test.py (hunk 2 above)
Comment (mnunberg) [reply src/pytest/test.py]: I've restored them

File: src/query_plan.c (hunk 3 above)
Comment (mnunberg) [reply src/query_plan.c]: Just a leftover

File: src/query_plan.c (hunk 6 above)
Comment (mnunberg) [reply src/query_plan.c]: Compilers will probably just inline it, but yes a macro is more appropriate.

**Pull request merged**